File: sdks/php/test/Api/FaxLineApiTest.php
<?php

declare(strict_types=1);

namespace Api;

use Dropbox\Sign\Api;
use Dropbox\Sign\Configuration;
use Dropbox\Sign\Model;
use Dropbox\Sign\Test\SignTestCase;
use Dropbox\Sign\Test\TestUtils;
use GuzzleHttp;
use SplFileObject;

class FaxLineApiTest extends SignTestCase
{
    /** @var Api\FaxApi */
    protected $api;

    protected function setUp(): void
    {
        parent::setUp();

        $this->client = new GuzzleHttp\Client([
            'handler' => GuzzleHttp\HandlerStack::create($this->handler),
        ]);

        $this->api = new Api\FaxLineApi(
            Configuration::getDefaultConfiguration(),
            $this->client,
        );
    }

    public function testFaxSend()
    {
        $requestClass = Model\FaxLineCreateRequest::class;
        $requestData = TestUtils::getFixtureData($requestClass)['default'];

        $responseClass = Model\FaxLineResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $obj = Model\FaxLineCreateRequest::init(
            $requestData
        );

        $response = $this->api->faxLineCreate($obj);
        $serialized = TestUtils::toArray($response);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, $serialized);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testFaxLineList()
    {
        $responseClass = Model\FaxLineListResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $response = $this->api->faxLineList();
        $serialized = TestUtils::toArray($response);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, $serialized);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testFaxLineGet()
    {
        $faxLineNumber = '14155557897';

        $responseClass = Model\FaxLineResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $response = $this->api->faxLineGet($faxLineNumber);
        $serialized = TestUtils::toArray($response);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, $serialized);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }
}

File: sdks/java-v1/src/test/java/com/dropbox/sign/api/FaxLineApiTest.java
package com.dropbox.sign.api;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.dropbox.sign.ApiClient;
import com.dropbox.sign.TestHelper;
import com.dropbox.sign.model.*;
import com.fasterxml.jackson.databind.JsonNode;
import org.junit.jupiter.api.Test;

public class FaxLineApiTest {
    @Test
    public void faxLineCreateTest() throws Exception {
        JsonNode expectedResponseData =
                TestHelper.getJsonContents(FaxLineResponse.class.getSimpleName());
        FaxLineResponse expectedResponse = FaxLineResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        JsonNode requestData =
                TestHelper.getJsonContents(FaxLineCreateRequest.class.getSimpleName());

        FaxLineCreateRequest request = FaxLineCreateRequest.init(requestData.toString());

        FaxLineApi api = new FaxLineApi(apiClient);
        FaxLineResponse response = api.faxLineCreate(request);

        assertEquals(expectedResponse, response);
    }

    @Test
    public void faxLineGetTest() throws Exception {
        String faxLineNumber = "14155557897";

        JsonNode expectedResponseData =
                TestHelper.getJsonContents(FaxLineResponse.class.getSimpleName());
        FaxLineResponse expectedResponse = FaxLineResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        FaxLineApi api = new FaxLineApi(apiClient);
        FaxLineResponse response = api.faxLineGet(faxLineNumber);

        assertEquals(expectedResponse, response);
    }

    @Test
    public void faxLineListTest() throws Exception {
        JsonNode expectedResponseData =
                TestHelper.getJsonContents(FaxLineListResponse.class.getSimpleName());
        FaxLineListResponse expectedResponse =
                FaxLineListResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        FaxLineApi api = new FaxLineApi(apiClient);
        FaxLineListResponse response = api.faxLineList();

        assertEquals(expectedResponse, response);
    }
}
